### ISSUE TITLE
How to return same js instance from a rust function?

### ISSUE THREAD

**Issue by richarddd** [DelSkayn/rquickjs#172]
Example:

```rust
fn update<'js>(&self, ctx: Ctx<'js>, value: Value<'js>) -> Result<Value<'js>> {
        let bytes = get_bytes(ctx, value)?;
        self.context.update(&bytes);
        self.into_js(ctx)
    }
```

`self.into_js(ctx)` creates a new instance. How can i return the original one?

**Comment by DelSkayn** [comment]
Instead of taking the self instances as `&self` you can retrieve the this value as its javascript object by taking it as `This<Class<'js,Self>>`. You can return that object without any problems.

**Comment by richarddd** [comment]
Any idea how would this be done using the new proc macro? :) Or do i need to manually implement the JsClass trait?

Example:
```rust
#[rquickjs::class]
#[derive(rquickjs::class::Trace)]
struct TestClass {
    #[qjs(skip_trace)]
    value: String,
}

#[rquickjs::methods]
impl TestClass {
    #[qjs(constructor)]
    pub fn new(value: String) -> Self {
        Self { value }
    }

    pub fn append<'js>(this: This<Class<'js, Self>>, data: String) -> Class<'js, Self> {
        this.0
    }
}
```

> the trait bound `__impl_append: JsClass<'js>` is not satisfied
> the following other types implement trait `JsClass<'js>`:
>   RustFunction<'js>
>   TestClass

Also tried with:
```rust
pub fn append<'js>(self, data: String) -> Self {
    self
}
```

Which results in:

> the trait bound `rquickjs::function::This<TestClass>: FromParam<'_>` is not satisfied
> the trait `FromParam<'js>` is implemented for `rquickjs::function::This<T>`

**Comment by DelSkayn** [comment]
This seems to be a bug in the macro, It works if you use the resolved type instead of Self.

The macro doesn't replace Self with the right type when implementing IntoJsFunc for the method.

**Comment by DelSkayn** [comment]
Should now be fixed in fd957b8.
